beets/autotag/hooks.py

```diff
@@ -529,14 +535,20 @@ def track_for_mbid(recording_id):
 def albums_for_id(album_id):
     """Get a list of albums for an ID."""
     candidates = [album_for_mbid(album_id)]
-    candidates.extend(plugins.album_for_id(album_id))
+    plugin_albums = plugins.album_for_id(album_id)
+    for a in plugin_albums:
+        plugins.send('trackinfo_received', info=a)
+    candidates.extend(plugin_albums)
```

Comment:
Since we have a loop in this function anyway, let's make it go over the entire `candidates` list, and eliminate the event in the special `_for_mbid` functions. A little less code for a little more clarity.